### PR TITLE
Autocomplete properties in `tween_property`

### LINKED ISSUE #6739: Add property autocompletion for `tween_property()`
### Describe the project you are working on

Game in Godot.

### Describe the problem or limitation you are having in your project

I use Tweens quite a bit, but I sometimes lack some more friendly way to tween properties. `tween_property()` does not come with autocompletion, so it's one of the very few things that you need to type manually.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

`tween_property()` should provide autocompletion for property list, similar to e.g. method list when using old `connect()`:
image
The list depends on the previous argument. It would appear here and list properties of the object in the first argument:
image

It would make it easier to pick properties to tween.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I think it would be implemented similarly to the aforementioned list of methods. I don't know if it's done via `get_argument_options()` or something different. I remember I tried to implement it myself a couple of times, but failed >_>

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Kind of? If you have no autocompletion, the workaround is to just write it I guess.

### Is there a reason why this should be core and not an add-on in the asset library?

Not sure if possible. Tween is core feature tho and it would benefit everyone that uses it.

### PR DESCRIPTION
**Pull request by HolonProduction**
Fixes godotengine/godot-proposals#6739

### CODE REVIEW

File: modules/gdscript/gdscript_editor.cpp [hunk 1]
```diff
@@ -28,13 +28,16 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
 /**************************************************************************/
 
+#include "core/object/script_language.h"
```
Comment (AThousandShips) [inline modules/gdscript/gdscript_editor.cpp:31]:
```suggestion
```

File: modules/gdscript/gdscript_editor.cpp [hunk 2]
```diff
@@ -28,13 +28,16 @@
...
 #include "gdscript.h"
 
 #include "gdscript_analyzer.h"
 #include "gdscript_compiler.h"
 #include "gdscript_parser.h"
 #include "gdscript_tokenizer.h"
 #include "gdscript_utility_functions.h"
+#include "modules/gdscript/gdscript_function.h"
+#include <cstddef>
```
Comment (AThousandShips) [inline modules/gdscript/gdscript_editor.cpp:40]:
```suggestion
```

File: modules/gdscript/gdscript_editor.cpp [hunk 3]
```diff
@@ -2598,6 +2601,64 @@ static void _find_call_arguments(GDScriptParser::CompletionContext &p_context, c
...
+				if (p_argidx == 1 && p_context.node && p_context.node->type == GDScriptParser::Node::CALL && ClassDB::is_parent_class(class_name, SNAME("Tween")) && p_method == SNAME("tween_property")) {
+					// Get tweened objects properties
+					GDScriptParser::ExpressionNode *tweened_object = static_cast<GDScriptParser::CallNode *>(p_context.node)->arguments[0];
+					StringName native_type = tweened_object->datatype.native_type;
+					switch (tweened_object->datatype.kind) {
+						case GDScriptParser::DataType::SCRIPT: {
+							Ref<Script> script = tweened_object->datatype.script_type;
+							native_type = script->get_instance_base_type();
+							int n = 0;
+							while(script.is_valid()) {
```
Comment (AThousandShips) [inline modules/gdscript/gdscript_editor.cpp]:
```suggestion
							while (script.is_valid()) {
```

File: modules/gdscript/gdscript_editor.cpp (hunk 2 above)
Comment (HolonProduction) [reply modules/gdscript/gdscript_editor.cpp]: Those damn IDE out includes

**Comment by HolonProduction** [comment]
> Autocompletion does not work well with custom classes: There are no properties from base native class.

Should handle inheritance correctly now.

File: modules/gdscript/gdscript_editor.cpp [hunk 4]
```diff
@@ -2598,6 +2598,64 @@ static void _find_call_arguments(GDScriptParser::CompletionContext &p_context, c
...
+							native_type = script->get_instance_base_type();
+							int n = 0;
+							while (script.is_valid()) {
+								List<PropertyInfo> properties;
+								script->get_script_property_list(&properties);
+								for (const PropertyInfo &E : properties) {
+									if (E.usage & (PROPERTY_USAGE_SUBGROUP | PROPERTY_USAGE_GROUP | PROPERTY_USAGE_CATEGORY | PROPERTY_USAGE_INTERNAL)) {
+										continue;
+									}
+									ScriptLanguage::CodeCompletionOption option(vformat("\"%s\"", E.name), ScriptLanguage::CODE_COMPLETION_KIND_PLAIN_TEXT, ScriptLanguage::CodeCompletionLocation::LOCATION_LOCAL + n);
```
Comment (KoBeWi) [inline modules/gdscript/gdscript_editor.cpp:2617]: You could use `CODE_COMPLETION_KIND_MEMBER` here. Same below.

**Review by KoBeWi — APPROVED** [review]
Works correctly on the editor side.

**Review by adamscott — APPROVED** [review]
Seems fine!

**Review by AThousandShips — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**